### ISSUE TITLE
Resource Health fine tuning lacking

### ISSUE THREAD

**Issue by stevedistef** [Azure/azure-monitor-baseline-alerts#208]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

One of my customers implemented AMBA GA on Production Subscriptions, and they are seeing a lot of ResourceHealthUnhealthyAlert.   We can see from the policy it can be turned off, but they would like to fine tune it (thresholding)?  Is that possible?
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/e147ad1b-334f-428d-b56d-d367b7e379e8)

When checking the alert itself, we see only these options. Is there any way to fine tune this alert?

 
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/2079e4ea-2182-4413-abbc-948da172036d)

**Comment by Brunoga-MS** [comment]
Hello @stevedistef ,
thanks for your feedback. Based on the UI provided by the Service / Resource Health alerts, it seems there are no options to further fine tune these alerts. The only possible fine tuning I can see here is the reduction of the statuses listed in the **_Previous resource status_** for which I need to investigate internally. However, one question came to mind: are these alerts fired for the same resource or for resources that are just named the same but located into different subscriptions or resource groups?

Thanks,
Bruno.

**Comment by stevedistef** [comment]
Buon Giorno @bruno and as thanks as always helping us adopt AMBA.   You and the whole tiger team you have there under @paulgrimley really make it much easier than a DIY project :-)

OK so on this one, as we discussed, I can also see this environment has these resource health alerts and I filtered for one of the resources which shows up alot, sometimes with the same time stamp.  I also filtered for only the last 30 days and then sorted by time:   
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/915eaee3-664c-4bb4-bd33-e21a2537fe60)

When we checked the two which seemed to be redundant, they are actually different (2 different alerts).
WHen we check the first one with that same timestamp, we see this:
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/8322d751-2f45-43a0-9fbe-93a75282967f)

WHen we check the other which came at the same time, we see this different alert:
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/04f83d71-b3fb-4498-98dc-c78474469610)

so the question becomes do we really need to see both....
When examining the actual alert, we see this:  (I clicked on Alert Rule in previous screen shot to get here):
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/8fcd7260-3bef-4441-b337-49dcfafc8477)

ANd then EDIT:
we see that perhaps we have set up too many "previous conditions":
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/ae458668-27b5-477f-8a5c-8e1f8399dd50)

I am going to ask the team using AMBA to go to Monito:ALerts:Alert Rules and edit the Resource Health alter for each of their subscriptions, removing the two previous conditions, and save it.
so repeating this step 4x in this case:
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/5528a0fa-733a-40ea-b270-8bd29d6e1066)

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/159309739/366d5e5f-cbb3-4e5a-bbb7-b6fe325aff1f)

We will see if this is acceptable....

**Comment by stevedistef** [comment]
Customer trying over the weekend!

**Comment by dbelso** [comment]
Hi All,
I tried the workaround and it seemed working for few days but now the issue got worse and flood our Monitoring page.
![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/107922097/92049b79-d1be-4910-92bb-39afca2b6172)

**Comment by Brunoga-MS** [comment]
Hello @stevedistef ,
from your communication it looks like the fine tuning we applied was partially working. At this point we need to investigate further to understand why this is happening. We will keep you posted.

Thanks,
Bruno.

**Comment by MarcoJanse** [comment]
I have a sort like question. First of all, I was wondering why the `ResourceHealthUnhealtyAlert` has a target scope of _All resources in `Subscription>`_. I was expecting the MonitorDisable parameter to disable the ResourceHealth alerts for all the resources depending on the tag value I had set. (like Dev or Sandbox).

The amount of events from ResourceHealth for just one VM that's being powered off is quite overwhelming.

There's even an alert when the status does not actually transition:

![Image](https://github.com/user-attachments/assets/06f8aa0b-d5f3-4903-836f-f5efd233dc3f)

For now, I have created a suppression rule for a couple of test VMs that are stopped/started frequently. 
Is there a better way to do this that I'm not seeing?

**Comment by Code-Gibbon** [comment]
> I have a sort like question. First of all, I was wondering why the `ResourceHealthUnhealtyAlert` has a target scope of _All resources in `Subscription>`_. I was expecting the MonitorDisable parameter to disable the ResourceHealth alerts for all the resources depending on the tag value I had set. (like Dev or Sandbox).
> 
> The amount of events from ResourceHealth for just one VM that's being powered off is quite overwhelming.
> 
> There's even an alert when the status does not actually transition:
> 
> ![Image](https://github.com/user-attachments/assets/06f8aa0b-d5f3-4903-836f-f5efd233dc3f)
> 
> For now, I have created a suppression rule for a couple of test VMs that are stopped/started frequently. Is there a better way to do this that I'm not seeing?

We are seeing similar in our deployments, especially in our AVD subscriptions. It seems the monitoring tag isn't respected by the AMBA deployed resource health alerts (are others seeing this?) so we are getting hit with a considerable number of resource health alerts. Is a suppression rule the best way to handle this or are there other things in the pipeline? Thanks :)

**Comment by riosengineer** [comment]
> > I have a sort like question. First of all, I was wondering why the `ResourceHealthUnhealtyAlert` has a target scope of _All resources in `Subscription>`_. I was expecting the MonitorDisable parameter to disable the ResourceHealth alerts for all the resources depending on the tag value I had set. (like Dev or Sandbox).
> > The amount of events from ResourceHealth for just one VM that's being powered off is quite overwhelming.
> > There's even an alert when the status does not actually transition:
> > ![Image](https://github.com/user-attachments/assets/06f8aa0b-d5f3-4903-836f-f5efd233dc3f)
> > For now, I have created a suppression rule for a couple of test VMs that are stopped/started frequently. Is there a better way to do this that I'm not seeing?
> 
> We are seeing similar in our deployments, especially in our AVD subscriptions. It seems the monitoring tag isn't respected by the AMBA deployed resource health alerts (are others seeing this?) so we are getting hit with a considerable number of resource health alerts. Is a suppression rule the best way to handle this or are there other things in the pipeline? Having the Resource Health alert respect the monitoring tag would be a big help.
> 
> Thanks :)

I have the same, it's not respected, although my assignments may be a month or two old now.

**Comment by Brunoga-MS** [comment]
Hello @Code-Gibbon and @riosengineer ,
thanks for your additional feedback. Did you create the _MonitorDisable_ tag after the first remediation? If so, you might need to delete the alerts for the given subscriptions. For Metrics and Activity Log alerts, the _MonitorDisable_ tag is only taken into consideration during policy evaluation. If a given policy has been evaluated, marked as non-compliant and then remediated, and then a resource is tagged, the alert(s) will not be removed. That might explain why you're still getting alerts even with the _MonitorDisable_ tag in place.

Differently if you put the tag in place before running the first remediation (say right after the solution deployment) it must be respected.

Hope that helps.

Thanks,
Bruno.

**Comment by riosengineer** [comment]
> Hello @Code-Gibbon and @riosengineer ,
> thanks for your additional feedback. Did you create the _MonitorDisable_ tag after the first remediation? If so, you might need to delete the alerts for the given subscriptions. For Metrics and Activity Log alerts, the _MonitorDisable_ tag is only taken into consideration during policy evaluation. If a given policy has been evaluated, marked as non-compliant and then remediated, and then a resource is tagged, the alert(s) will not be removed. That might explain why you're still getting alerts even with the _MonitorDisable_ tag in place.
> 
> Differently if you put the tag in place before running the first remediation (say right after the solution deployment) it must be respected.
> 
> Hope that helps.
> 
> Thanks,
> Bruno.

Thanks Bruno, I think this explains it as I had added some post deployment. Presumably this is the same for other exemptions like VM disks or other tag overrides?

**Comment by Brunoga-MS** [comment]
Yes and no @riosengineer . When tags (MonitorDisable, Logical Volumes Exclusion and Override alert thresholds tags) are used inside Log-search alerts (like Azure VMs, Hybrid VMs) they are honored by the query, so you can add the tags after the remediation and they will be respected. For ActivityLog and Metric alerts, you need to delete the "unnecessary" alerts after tagging. The tag will only ensure the "unnecessary" alerts will not be recreated by the remediation script the next time it is executed.

Hope that clarify more.

**Comment by Brunoga-MS** [comment]
BTW @riosengineer , other than using the MonitorDisable tag that will impact also other alerts targeted at the same resource you might want to consider the exclusion feature documented at https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/HowTo/Exclude_resources_from_policy_assignment/ and https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/Resources/FAQ/#can-i-exclude-management-groups-or-subscriptions-from-policy-assignment. For existing alerts, this should be used used together with the alert removal.

**Comment by riosengineer** [comment]
> BTW @riosengineer , other than using the MonitorDisable tag that will impact also other alerts targeted at the same resource you might want to consider the exclusion feature documented at https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/HowTo/Exclude_resources_from_policy_assignment/ and https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/Resources/FAQ/#can-i-exclude-management-groups-or-subscriptions-from-policy-assignment. For existing alerts, this should be used used together with the alert removal.

Yeah good shout. It's mostly my fault anyway😄, I was testing a VM deployment for a new prod setup and AMBA enrolled and started alerting, so I tried the disable tag post deployment of the alerts in a new sub.

**Comment by Brunoga-MS** [comment]
Since the resource health fine tuning is not something we can drive from the AMBA-ALZ perspective but it's more related to the platform, I will close this issue. Should you have any further issue with Resource Health (and not with the way it is implemented in AMBA-ALZ) please work with the Microsoft Support.

Thanks,
Bruno.
